src/Range.jsx

```diff
@@ -176,10 +176,17 @@ class Range extends React.Component {
     const { marks, step, min, max } = this.props;
     const cache = this._getPointsCache;
     if (!cache || cache.marks !== marks || cache.step !== step) {
+      const decimal = step.toString().split('.')[1] && step.toString().split('.')[1].length;
       const pointsObject = { ...marks };
       if (step !== null) {
-        for (let point = min; point <= max; point += step) {
+        let point = min;
+        while (point <= max) {
```

Comment:
Another idea,  what if when `step` is very small compare to `max - min`, we don't calculate the whole points array at first, instead just get the formula，something like： `value = f(index)`, then the cost just happen when every time getter